rotkehlchen/chain/avalanche/manager.py

```diff
@@ -107,40 +107,34 @@ def get_code(self, account: ChecksumEthAddress) -> str:
 
     def get_transaction_receipt(
             self,
-            tx_hash: str,
+            tx_hash: EVMTxHash,
     ) -> Dict[str, Any]:
         tx_receipt = self.covalent.get_transaction_receipt(tx_hash)
         if tx_receipt is None:
             tx_receipt = self.w3.eth.get_transaction(tx_hash).__dict__  # type: ignore
+            tx_receipt['hash'] = EVMTxHash(tx_receipt['hash'])  # web3 returns HexBytes
             return tx_receipt
         try:
-            # Turn hex numbers to int
-            tx_receipt.pop('from_address_label', None)
-            tx_receipt.pop('to_address_label', None)
-            block_number = tx_receipt['block_height']
-            tx_receipt['blockNumber'] = tx_receipt.pop('block_height', None)
-            tx_receipt['cumulativeGasUsed'] = tx_receipt.pop('gas_spent', None)
-            tx_receipt['gasUsed'] = tx_receipt['cumulativeGasUsed']
-            successful = tx_receipt.pop('successful', None)
-            tx_receipt['status'] = 1 if successful else 0
-            tx_receipt['transactionIndex'] = 0
-            txhash = tx_receipt.pop('tx_hash')
-            tx_receipt['hash'] = txhash
-
-            # TODO input and nonce is decoded in Covalent api, encoded in future
-            tx_receipt['input'] = '0x'
-            tx_receipt['nonce'] = 0
-            for index, receipt_log in enumerate(tx_receipt['log_events']):
-                receipt_log['blockNumber'] = block_number
+            result_receipt: Dict[str, Any] = {
+                'hash': deserialize_evm_tx_hash(tx_receipt['tx_hash']),
+                'blockNumber': tx_receipt['block_height'],
+                'from': to_checksum_address(tx_receipt['from_address']),
```

Comment:
There is a big difference between the code you add and the code you removed.

The guy before you was doing `pop` with `None` as default. This would mean that perhaps some times the key can be missing from the response. Can it? If it can then your code will fail with `KeyError`.